src/lib.rs

```diff
@@ -1,565 +1,525 @@
 //! # libcapstone.so.3 bindings
 //!
 //! If you want to compile this for another target,  `wasm32-unknown-emscripten`, for example,
-//! it is currently recommended to pass the feature flag `build_src_cmake` to
+//! it is currently recommended to pass the feature flag `use_bundled_capstone_cmake` to
 //! build capstone using the cmake build system (which requires cmake to be installed).
 //!
-//! This has seen some (limited) testing and has been seen to work on the `wasm32-unknown-emscripten` target at least.
+//! This has seen some (limited) testing and has been seen to work on the
+//! `wasm32-unknown-emscripten` target at least.
 //!
 //! Compiling on windows has not been tested, although this should be easy to setup.
+//!
+//! The following architectures are supported:
+//!
+//! * `arm`: ARM
+//! * `arm64`: ARM64 (also known as AArch64)
+//! * `mips`: MIPS
+//! * `ppc`: PowerPC
+//! * `sparc`: SPARC
+//! * `sysz`: System z
+//! * `x86`: x86 family (includes 16, 32, and 64 bit modes)
+//! * `xcore`: XCore
+//!
+//! For each architecture, *at least* the following types are defined (replace `ARCH` with
+//! architecture names shown above):
+//!
+//! * `enum ARCH_insn`: instruction ids
+//! * `enum ARCH_insn_group`: architecture-specific group ids
+//! * `enum ARCH_op_type`: instruction operand types ids
+//! * `enum ARCH_reg`<sup>1</sup>: register ids
+//! * `struct ARCH_op_mem`: operand referring to memory
+//! * `struct cs_ARCH_op`: instruction operand
+//! * `struct cs_ARCH`: instruction
+//!
+//! <sup>1</sup>: "constified" enum because discriminant values are not unique
 
+// Suppress errors from Capstone names
+#![allow(non_upper_case_globals)]
 #![allow(non_camel_case_types)]
-extern crate libc;
```

Comment:
I figured that if you prefer, you could make a new "bindgen-wip" branch. Once you felt that it was in good enough condition, you could merge the changes to master.

It would be confusing to a consumer of this library that the interface is different based on which features are specified.